docs/configuration/http_conn_man/rds.rst

```diff
@@ -28,7 +28,7 @@ route_config_name
   *(required, string)* The name of the route configuration. This name will be passed to the
   :ref:`RDS HTTP API <config_http_conn_man_rds_api>`. This allows an Envoy configuration with
   multiple HTTP listeners (and associated HTTP connection manager filters) to use different route
-  configurations.
+  configurations. The route_config_name must **not** contain ``:``.
```

Comment:
Thinking about this some more, I was being near sighted, and thought that it was possible to fall into the same collision situation that I was trying to escape. But that is not the case if one of the strings has the blocked character. 

@htuch how serious are we thinking about relaxing the `:` constraint? I could leave a TODO to move the to pair, or struct key when that happens, but for now I'll leave it this way to avoid having to deal with hashing the key.